types/utils.go

```diff
@@ -0,0 +1,23 @@
+package types
+
+import "github.com/prysmaticlabs/prysm/shared/bls"
+
+type HashTreeRoot interface {
+	HashTreeRoot() ([32]byte, error)
+}
+
+func VerifySignature(obj HashTreeRoot, pk, s []byte) (bool, error) {
```

Comment:
It would be much better to define the signing API in a way where you are forced to include a signing domain and fork version etc. Signatures should be constructed in a way where they are unique for the purpose of the work and version of the protocol. See `compute_signing_root` and `get_domain` in the phase0 beacon spec. You essentially want to compute a "domain" (32 bytes) from a fork digest (constructed from genesis + fork info) and a type-domain (e.g. proposals and attestations have different domains to be sure they never have the same signature for two different intents). Or you can implement your own non-standard domain method. But definitely don't just create a BLS signature over data that might have a different meaning later on in the protocol. cc @lightclient

Reply:
Also relevant: https://github.com/ethereum/consensus-specs/pull/2884 for the domain

Reply:
The current validation code is already merged, in this PR it's just moved. Maybe we should transfer this into it's own issue and make an update independent to this PR, which just complements the code currently in master.

Reply:
👉  https://github.com/protolambda/mergemock/issues/29